src/ui/public/saved_object_table/item_table_state.js

```diff
@@ -0,0 +1,38 @@
+import { SortOrder } from 'ui_framework/components/table/sort_order';
+
+export class ItemTableState {
```

Comment:
For example, if we wrote `item_selector.js` like this:

```javascript
export class ItemSelector {
  constructor(ids) {
    this.idToIsSelectedMap = {};
    this.ids = [];

    this.update(ids);
  }

  update(ids) {
    this.ids = ids;
    const idToIsSelectedMap = {};
    ids.forEach(id => idToIsSelectedMap[id] = this.idToIsSelectedMap[id]);
    this.idToIsSelectedMap = idToIsSelectedMap;
  }

  deselectAll() {
    this.idToIsSelectedMap = {};
  }

  selectAll() {
    this.ids.forEach(id => this.idToIsSelectedMap[id] = true);
  }

  areAllItemsSelected() {
    const isSelectedValues = this.ids.map(id => this.idToIsSelectedMap[id]);
    return !isSelectedValues.includes(false);
  }

  toggleAll() {
    if (itemTableState.areAllItemsSelected()) {
      this.deselectAll(itemTableState);
    } else {
      this.selectAll(itemTableState);
    }
  }

  selectItem(id) {
    this.idToIsSelectedMap[id] = true;
    itemTableState.selectedItems.push(item);
  }

  deselectItem(id) {
    delete this.idToIsSelectedMap[id];
  }

  isItemSelected(id) {
    return this.idToIsSelectedMap[id];
  }

  getSelectedItemsCount() {
    return this.ids.filter(id => this.idToIsSelectedMap[id]).length;
  }
}
```

Then this module encapsulates the entire concern of item-selection. Instead of spreading this logic between `item_table_state.js` and `item_sorter.js`, it's located in one file. It also simplifies its dependencies. Now it only needs to know about primitive values (ids and booleans) instead of a complex state object (itemTableState). This makes it easier to understand, reason about, and test.